src/pages/resultsView/plots/PlotsTab.tsx

```diff
@@ -3459,24 +3457,28 @@ export default class PlotsTab extends React.Component<IPlotsTabProps, {}> {
     }
 
     @computed get selectedGenericAssayEntitiesGroupByGenericAssayTypeFromUrl() {
-        const result = _.reduce(
-            this.props.store.selectedGenericAssayEntities,
-            (acc, value, key) => {
-                if (
-                    this.props.store.molecularProfileIdSuffixToMolecularProfiles
-                        .result[key]
-                ) {
-                    const type = this.props.store
-                        .molecularProfileIdSuffixToMolecularProfiles.result[
-                        key
-                    ][0].genericAssayType;
-                    acc[type] = acc[type] ? _.union(value, acc[type]) : value;
-                    return acc;
-                }
-            },
-            {} as { [genericAssayType: string]: string[] }
-        );
-        return result;
+        if (this.props.store.molecularProfileIdToMolecularProfile.isComplete) {
+            const result = _.reduce(
+                this.props.store.selectedGenericAssayEntities,
+                (acc, value, key) => {
+                    if (
+                        this.props.store.molecularProfileIdToMolecularProfile
+                            .result[key]
+                    ) {
+                        const type = this.props.store
+                            .molecularProfileIdToMolecularProfile.result[key]
+                            .genericAssayType;
+                        acc[type] = acc[type]
+                            ? _.union(value, acc[type])
+                            : value;
+                        return acc;
+                    }
```

Comment:
seems like this reduce callback doesn't always return acc? i'm suprisied this doesn't have ts error.  maybe i'm misreading